sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -220,10 +243,11 @@ private[jdbc] class JDBCRDD(
     partitions: Array[Partition],
     url: String,
     options: JDBCOptions,
-    groupByColumns: Option[Array[String]],
-    sample: Option[TableSampleInfo],
-    limit: Int,
-    sortOrders: Array[SortOrder])
+    groupByColumns: Option[Array[String]] = None,
+    sample: Option[TableSampleInfo] = None,
+    limit: Int = 0,
+    sortOrders: Array[SortOrder] = Array.empty[SortOrder],
+    predicates: Array[Predicate] = Array.empty[Predicate])
```

Comment:
We can translate v1 `Filter` to v2 `Predicate`, to simplify the code. Then the JDBC implementation only need to hold and deal with v2 `Predicate`.

Reply:
OK